tests/postgres/select_test.go

```diff
@@ -658,6 +660,98 @@ ORDER BY city.city_id, address.address_id, customer.customer_id;
 `)
 }
 
+// Test join with sql.NullInt64 primary keys
+func TestExecution5(t *testing.T) {
+
+	var dest []struct {
+		CityID   sql.NullInt64 `sql:"primary_key" alias:"city.city_id"`
+		CityName string        `alias:"city.city"`
+
+		Customers []struct {
+			CustomerID sql.NullInt64 `sql:"primary_key" alias:"customer_id"`
+			LastName   *string       `alias:"last_name"`
+
+			Address struct {
+				AddressID   sql.NullInt64 `sql:"primary_key" alias:"AddressId"`
+				AddressLine string        `alias:"address.address"`
+			} `alias:"address.*"`
+		} `alias:"customer"`
+	}
+
+	stmt := City.
+		INNER_JOIN(Address, Address.CityID.EQ(City.CityID)).
+		INNER_JOIN(Customer, Customer.AddressID.EQ(Address.AddressID)).
+		SELECT(
+			City.CityID,
+			City.City,
+			Customer.CustomerID,
+			Customer.LastName,
+			Address.AddressID,
+			Address.Address,
+		).
+		WHERE(City.City.EQ(String("London")).OR(City.City.EQ(String("York")))).
+		ORDER_BY(City.CityID, Address.AddressID, Customer.CustomerID)
+
+	testutils.AssertDebugStatementSql(t, stmt, `
+SELECT city.city_id AS "city.city_id",
+     city.city AS "city.city",
+     customer.customer_id AS "customer.customer_id",
+     customer.last_name AS "customer.last_name",
+     address.address_id AS "address.address_id",
+     address.address AS "address.address"
+FROM dvds.city
+     INNER JOIN dvds.address ON (address.city_id = city.city_id)
+     INNER JOIN dvds.customer ON (customer.address_id = address.address_id)
+WHERE (city.city = 'London'::text) OR (city.city = 'York'::text)
+ORDER BY city.city_id, address.address_id, customer.customer_id;
+`, "London", "York")
+
+	err := stmt.Query(db, &dest)
+
+	require.NoError(t, err)
+	require.Equal(t, len(dest), 2)
+	testutils.AssertJSON(t, dest, `
+[
```

Comment:
It seems that tests fails because, json doesn't match. Expected json:

```json
[
	{
		"CityID": {
			"Int64": 312,
			"Valid": true
		},
		"CityName": "London",
		"Customers": [
			{
				"CustomerID": {
					"Int64": 252,
					"Valid": true
				},
				"LastName": "Hoffman",
				"Address": {
					"AddressID": {
						"Int64": 256,
						"Valid": true
					},
					"AddressLine": "1497 Yuzhou Drive"
				}
			},
			{
				"CustomerID": {
					"Int64": 512,
					"Valid": true
				},
				"LastName": "Vines",
				"Address": {
					"AddressID": {
						"Int64": 517,
						"Valid": true
					},
					"AddressLine": "548 Uruapan Street"
				}
			}
		]
	},
	{
		"CityID": {
			"Int64": 589,
			"Valid": true
		},
		"CityName": "York",
		"Customers": [
			{
				"CustomerID": {
					"Int64": 497,
					"Valid": true
				},
				"LastName": "Sledge",
				"Address": {
					"AddressID": {
						"Int64": 502,
						"Valid": true
					},
					"AddressLine": "1515 Korla Way"
				}
			}
		]
	}
]
```

Reply:
Yes, it was silly on my part. I am so used to `null.Int` by `github.com/guregu/null` that I forgot about the JSON marshaling of `sql.NullInt64`. I also added another test by using the aforementioned `null.Int`.

I was able to configure the environment to run some test, even tough not everything works. I had to pass trough some hoops to make it work on my M1 Mac. In my case `platform: linux/amd64` was needed in `docker-compose.yaml`.

Reply:
Yeah, I've heard M1 Mac tends to have problems with docker. Maybe we can include your `docker-compose.yaml` changes. M1 is getting popular, someone else might stumble on the same issue. It can be a separate PR.